src/main/kotlin/calculator/DelimiterExtractor.kt

```diff
@@ -0,0 +1,24 @@
+package calculator
+
+object DelimiterExtractor {
+    private val DEFAULT_DELIMITER_REGEX = "[,:]".toRegex()
+    private val CUSTOM_DELIMITER_EXTRACT_REGEX = Regex("//(.)\n(.*)")
+
+    fun extractNumbers(input: String): List<PositiveNumber> {
+        require(input.isNotEmpty()) { "입력 값이 비어있을 수는 없어요." }
+
+        val tokens = extractTokens(input)
+        return tokens.map { token -> PositiveNumber.of(token) }
+    }
+
+    private fun extractTokens(input: String): List<String> {
+        val result = CUSTOM_DELIMITER_EXTRACT_REGEX.find(input)
+        result?.let {
+            val customDelimiter = it.groupValues[1]
+            return it.groupValues[2].split(customDelimiter)
+        }
+
+        return input.split(DEFAULT_DELIMITER_REGEX)
```

Comment:
아래 처럼 활용해볼 수도 있겠네요 :)
```suggestion
        return CUSTOM_DELIMITER_EXTRACT_REGEX.find(input)
            ?.let {
                val customDelimiter = it.groupValues[1]
                it.groupValues[2].split(customDelimiter)
            }
            ?: input.split(DEFAULT_DELIMITER_REGEX)
```

Reply:
제안 주신 방식과 유사하게 리턴하고 싶었는데, 이런 방법이 있었군요!🙏